zerver/lib/events.py

```diff
@@ -354,6 +354,10 @@ def fetch_initial_state_data(
                 realm.demo_organization_scheduled_deletion_date
             )
 
+        # Presence system parameters for client behavior.
+        state["server_presence_ping_interval_seconds"] = settings.PRESENCE_PING_INTERVAL_SECS
+        state["server_presence_offline_threshold_seconds"] = settings.OFFLINE_THRESHOLD_SECS
```

Comment:
My understanding of the discussion at https://chat.zulip.org/#narrow/stream/378-api-design/topic/presence.20rewrite/near/1506221 was that the third param `server_presence_idle_timeout_seconds` should be left hard-coded in the app, so didn't add that here.

Reply:
Yes, that seems correct to me.